R/submit.R

```diff
@@ -10,20 +10,28 @@
 ##'
 ##' @param ... Disallowed additional arguments, don't use.
 ##'
+##' @param resources A list generated by `hipercow_resources` giving
+##'   the cluster resource requirements to run your task.
+##'
 ##' @param driver The name of the driver to use, or you can leave
 ##'   blank if only one is configured (this will be typical).
 ##'
 ##' @param root The hipercow root
 ##'
 ##' @export
-task_submit <- function(id, ..., driver = NULL, root = NULL) {
+task_submit <- function(id, ..., resources = NULL, 
+                        driver = NULL, root = NULL) {
   if (...length() > 0) {
     cli::cli_abort("Additional arguments to 'task_submit' not allowed")
   }
   root <- hipercow_root(root)
-
   dat <- hipercow_driver_prepare(driver, root, environment())
-
+  if (!is.null(resources$hold_until$computed)) {
```

Comment:
I feel this could be done at validation, unless someone sits on their resources a long time.

Reply:
I am in two minds about this. I wonder if people might create a couple of resource objects for their "test_job" and their "real_job" and their "massive_job_that_always_waits_til_midnight", and then have those as kind of ready-made resources to launch their jobs in the same way as last time, without having to think about it.

I am not sure whether this is terribly likely, and might not be totally advisable - once in a while, available resources might change (but then they will need a new package version and will have to rerun their resource things anyway).

 But also, I'm not sure doing the translation during submission causes any real problem does it?

Reply:
the only "problem" caused by doing it at submission is that there's two places to do the validation really.  In particular, the validation here must never fail or it will be quite annoying for the user.  We can see how this goes in practice if you want - I expect this flow will need some adjustment when it comes in contact with users, no matter what we pick.

We might sketch out a flow of validation etc as we complicate this too (environment variables and parallel configuration will all interact here too)